jkube-kit/resource/helm/src/test/java/org/eclipse/jkube/kit/resource/helm/HelmServiceTest.java

```diff
@@ -213,6 +213,33 @@ void generateHelmCharts_withValidValuesYamlFragment_usesMergedValues() throws Ex
       );
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  void generateHelmCharts_whenInvoked_thenGeneratedValuesYamlInAlphabeticalOrder() throws IOException {
+    // Given
+    resourceServiceConfig = ResourceServiceConfig.builder().resourceDirs(Collections.singletonList(
+            new File(Objects.requireNonNull(getClass().getResource("/valid-helm-fragments")).getFile())))
+        .build();
+    helmConfig.types(Collections.singletonList(HelmType.KUBERNETES));
+
+    // When
+    new HelmService(jKubeConfiguration, resourceServiceConfig, new KitLogger.SilentLogger())
+        .generateHelmCharts(helmConfig.build());
+
+    // Then
+    final Map<String, ?> savedValues = Serialization.unmarshal(helmOutputDirectory.resolve("kubernetes").resolve("values.yaml").toFile(), Map.class);
+    assertThat(savedValues.keySet())
+        .containsExactly("image", "ingress", "replaceableProperty", "replicaCount", "service", "serviceAccount");
+    assertThat(((Map<String, ?>) savedValues.get("ingress")).keySet())
+        .containsExactly("annotations", "className", "enabled", "hosts", "tls");
+    assertThat(((Map<String, ?>) savedValues.get("image")).keySet())
+        .containsExactly("pullPolicy", "repository", "tag");
+    assertThat(((Map<String, ?>) savedValues.get("serviceAccount")).keySet())
+        .containsExactly("annotations", "create", "name");
+    assertThat(((Map<String, ?>) savedValues.get("service")).keySet())
+        .containsExactly("port", "type");
```

Comment:
I like the new solution, efficient and simple.

However, the test doesn't completely prove it works.

It might be better to feed a specifically crafted `values.yaml` file in the `Given` section with unsorted, nested values.

And then verify the generated `values.yaml` file text contents (not deserializing it) in the `Then` section.